File: internal/service/testdata/matrix/basic.ts
import { readFile } from "fs";

interface Named {
  name: string;
}

type ID = string;

enum Color {
  Red,
  Blue,
}

class Greeter {
  hello(who: string): string {
    return "hi " + who;
  }
}

const counter = 0;

function helper(): number {
  return 1;
}

function main(): void {
  helper();
  helper();
}

File: internal/service/testdata/matrix/basic.java
import java.util.List;

public class Greeter {
  private int count = 0;

  public Greeter() {}

  public String hello(String who) {
    int local = 1;
    return "hi " + who + local;
  }

  public void run() {
    hello("world");
  }
}

interface Named {
  String name();
}

enum Color { RED, BLUE }

record Point(int x, int y) {}
